src/components/Card/Card.tsx

```diff
@@ -0,0 +1,52 @@
+import * as React from 'react';
+import {useNavigate} from 'react-router-dom';
+import {ListItemColumn, Teams, UserData} from 'types/types';
+import {CardContainer} from './Card.styled';
+
+interface CardProps {
+    id: string;
+    url?: string;
+    columns: ListItemColumn[];
+    hasNavigation?: boolean;
+    navigationProps?: UserData | Teams;
+}
+
+const Card = (props: CardProps): JSX.Element => {
+    const {
+        id,
+        columns,
+        url =  '',
+        hasNavigation = true,
+        navigationProps = null,
+    } = props;
+
+    const navigate = useNavigate();
+
+    const handleOnClick = (event: React.MouseEvent<HTMLDivElement>): void => {
+        event.preventDefault();
+
+        if (hasNavigation) {
+            navigate(url, {
+                state: navigationProps,
+            });
+        }
+    };
+
+    return (
+        <CardContainer
+            data-testid={`cardContainer-${id}`}
+            hasNavigation={hasNavigation}
+            onClick={handleOnClick}
+        >
+            {
+                columns.map(({key: columnKey, value}, index) => (
+                    <p data-testid='card-column' key={`${columnKey}-${index}`}>
```

Comment:
Added card-column as data-testid to be able to find this elements and count how many exists in the tests.